c2corg_ui/views/profile.py

```diff
@@ -0,0 +1,106 @@
+import logging
+
+from pyramid.renderers import render
+from pyramid.view import view_config
+from pyramid.httpexceptions import (
+    HTTPNotFound, HTTPInternalServerError)
+
+from c2corg_ui.views.document import Document
+from c2corg_ui.caching import cache_document_detail
+
+log = logging.getLogger(__name__)
+
+
+class Profile(Document):
+
+    _API_ROUTE = 'profiles'
+
+    @view_config(route_name='profiles_view')
+    def profile(self):
+        id, lang = self._validate_id_lang()
+
+        def render_page(profile, locales):
+            locale = list(filter(lambda l: l['lang'] == lang, locales))
+            if not locale:
+                raise HTTPNotFound('Requested lang does not exist')
+
+            self.template_input.update({
+                'lang': lang,
+                'profile': profile,
+                'locale': locale[0],
+                'locales': locales
+            })
+
+            return render(
+                'c2corg_ui:templates/profile/view.html',
+                self.template_input,
+                self.request
+            )
+
+        def load_data(old_api_cache_key=None):
+            not_modified, api_cache_key, document_and_locale = \
+                self._get_profile(id, lang, old_api_cache_key)
+            return not_modified, api_cache_key, document_and_locale
+
+        return self._get_or_create(
+            (id, lang), cache_document_detail, load_data, render_page,
+            self._get_cache_key)
+
+    def _get_profile(self, id, lang, old_api_cache_key=None):
```

Comment:
Maybe rename this method to `_get_profile_info`, because you are not requesting the profile.